lib/date_time/date_time_modem.c

```diff
@@ -215,7 +215,13 @@ void date_time_modem_store(struct tm *ltm, int tz)
 	}
 }
 
-void date_time_modem_xtime_subscribe(void)
+#define DATE_TIME_XTIME_SUBSCRIBE_STACK_SIZE 1024
+#define DATE_TIME_XTIME_SUBSCRIBE_PRIORITY 5
+K_THREAD_STACK_DEFINE(date_time_modem_xtime_subscribe_stack, DATE_TIME_XTIME_SUBSCRIBE_STACK_SIZE);
+
+static struct k_work_q date_time_modem_xtime_subscribe_work_q;
```

Comment:
Need to check if `AT+CCLK=?` should be handled as well

Reply:
Having a work queue for `%XTIME` and a thread for the `+CCLK` query seems excessive. Is there any reason why `date_time_update_thread()` couldn't be replaced with a delayable work running in the new work queue?

Reply:
Converted existing thread to work queue and combined them